asmcomp/riscv/proc.ml

```diff
@@ -38,8 +38,7 @@ let word_addressed = false
     s2-s9        8-15      arguments/results (preserved by C)
     t2-t6        16-20     temporary
     s0           21        general purpose (preserved by C)
```

Comment:
isn't s0 for the OCaml stack now and not fully general purpose?

Reply:
Hi Anil, as far as I remember, we use `s0` in only one spot as an auxiliary register to preserve the OCaml stack pointer (here: https://github.com/nojb/ocaml/blob/71b9ebae167c5574b25d18375df5c45715a0d787/asmcomp/riscv/emit.mlp#L326-L335) and to do that we mark `s0` as being killed by `noalloc` calls (here: https://github.com/nojb/ocaml/blob/71b9ebae167c5574b25d18375df5c45715a0d787/asmcomp/riscv/proc.ml#L244-L249). But as far as the register allocator is concerned I think that `s0` is still treated as a general purpose.

Perhaps some trick could be used to avoid the need to reserve `s0` in this way, but I couldn't figure it out so far...

Reply:
Ah yes, quite right, thanks.